### PR TITLE
Added support for Doctrine\Cache 1.7 with deleteMultiple

### LINKED ISSUE #2: Better implementation for Multi-Delete
Following on from discussion: https://github.com/Roave/DoctrineSimpleCache/pull/1#discussion_r94380168

Doctrine will soon support an interface for multi-delete apparently, so we can simplify and tidy up our implementation.

### PR DESCRIPTION
**Pull request by asgrim**
Fixes #2

### CODE REVIEW

File: src/Exception/CacheException.php [hunk 1]
```diff
@@ -8,26 +8,11 @@
 
 final class CacheException extends \RuntimeException implements PsrCacheException
 {
-    public static function fromNonClearableCache(DoctrineCache $cache) : self
+    public static function fromNonMultiOperationCache(DoctrineCache $cache) : self
```
Comment (Ocramius) [inline src/Exception/CacheException.php]: Minor BC break: requires `UPGRADE.md` notes or revert + deprecation

File: src/Exception/CacheException.php [hunk 2]
```diff
@@ -8,26 +8,11 @@
...
     {
         return new self(sprintf(
-            'The given cache %s was not clearable, but you tried to use a feature that requires a clearable cache.',
-            get_class($cache)
-        ));
-    }
-
-    public static function fromNonMultiGetCache(DoctrineCache $cache) : self
```
Comment (Ocramius) [inline src/Exception/CacheException.php]: Minor BC break: requires `UPGRADE.md` notes or revert + deprecation

File: src/Exception/CacheException.php [hunk 3]
```diff
@@ -8,26 +8,11 @@
...
-    {
-        return new self(sprintf(
-            'The given cache %s cannot multi-get, but you tried to use a feature that requires a multi-get cache.',
-            get_class($cache)
-        ));
-    }
-
-    public static function fromNonMultiPutCache(DoctrineCache $cache) : self
```
Comment (Ocramius) [inline src/Exception/CacheException.php]: Minor BC break: requires `UPGRADE.md` notes or revert + deprecation

File: test/asset/NotMultiPuttableCache.php [hunk 4]
```diff
@@ -1,12 +0,0 @@
-<?php
-declare(strict_types = 1);
-
-namespace RoaveTestAsset\DoctrineSimpleCache;
-
-use Doctrine\Common\Cache\Cache;
-use Doctrine\Common\Cache\ClearableCache;
-use Doctrine\Common\Cache\MultiGetCache;
-
-interface NotMultiPuttableCache extends Cache, ClearableCache, MultiGetCache
```
Comment (Ocramius) [inline test/asset/NotMultiPuttableCache.php]: Minor BC break: requires `UPGRADE.md` notes or revert + deprecation

File: src/Exception/CacheException.php (hunk 1 above)
Comment (asgrim) [reply src/Exception/CacheException.php]: `fromNonClearableCache` shouldn't have gone, oops. Added that back in :)

**Comment by asgrim** [comment]
All other BC breaks now documented in upgrade notes in `README.md`.

**Pull request merged**